spring-core/src/main/java/org/springframework/core/retry/RetryTemplate.java

```diff
@@ -0,0 +1,178 @@
+ * @see BackOffPolicy
+ * @see RetryListener
+ */
+public class RetryTemplate implements RetryOperations {
+
+	protected final LogAccessor logger = new LogAccessor(LogFactory.getLog(getClass()));
+
+	protected RetryPolicy retryPolicy = new MaxRetryAttemptsPolicy();
+
+	protected BackOffPolicy backOffPolicy = new FixedBackOffPolicy(1000);
+
+	protected RetryListener retryListener = new RetryListener() {
+	};
+
+	/**
+	 * Create a new retry template with default settings.
+	 */
+	public RetryTemplate() {
+	}
+
+	/**
+	 * Create a new retry template with a custom {@link RetryPolicy}.
+	 * @param retryPolicy the retry policy to use
+	 */
+	public RetryTemplate(RetryPolicy retryPolicy) {
+		Assert.notNull(retryPolicy, "Retry policy must not be null");
+		this.retryPolicy = retryPolicy;
+	}
+
+	/**
+	 * Create a new retry template with a custom {@link RetryPolicy} and {@link BackOffPolicy}.
+	 * @param retryPolicy the retry policy to use
+	 * @param backOffPolicy the backoff policy to use
+	 */
+	public RetryTemplate(RetryPolicy retryPolicy, BackOffPolicy backOffPolicy) {
+		this(retryPolicy);
+		Assert.notNull(backOffPolicy, "BackOff policy must not be null");
+		this.backOffPolicy = backOffPolicy;
+	}
+
+	/**
+	 * Set the {@link RetryPolicy} to use. Defaults to <code>MaxAttemptsRetryPolicy()</code>.
+	 * @param retryPolicy the retry policy to use. Must not be <code>null</code>.
+	 */
+	public void setRetryPolicy(RetryPolicy retryPolicy) {
+		Assert.notNull(retryPolicy, "Retry policy must not be null");
+		this.retryPolicy = retryPolicy;
+	}
+
+	/**
+	 * Set the {@link BackOffPolicy} to use. Defaults to <code>FixedBackOffPolicy(Duration.ofSeconds(1))</code>.
+	 * @param backOffPolicy the backoff policy to use. Must not be <code>null</code>.
+	 */
+	public void setBackOffPolicy(BackOffPolicy backOffPolicy) {
+		Assert.notNull(backOffPolicy, "BackOff policy must not be null");
+		this.backOffPolicy = backOffPolicy;
+	}
+
+	/**
+	 * Set the {@link RetryListener} to use. Defaults to a <code>NoOp</code> implementation.
+	 * If multiple listeners are needed, use a {@link CompositeRetryListener}.
+	 * @param retryListener the retry listener to use. Must not be <code>null</code>.
+	 */
+	public void setRetryListener(RetryListener retryListener) {
+		Assert.notNull(retryListener, "Retry listener must not be null");
+		this.retryListener = retryListener;
+	}
+
+	/**
+	 * Call the retry callback according to the configured retry and backoff policies.
+	 * If the callback succeeds, its result is returned. Otherwise, a {@link RetryException}
+	 * will be thrown to the caller having all attempt exceptions as suppressed exceptions.
+	 * @param retryCallback the callback to call initially and retry if needed
+	 * @param <R> the type of the result
+	 * @return the result of the callback if any
+	 * @throws RetryException thrown if the retry policy is exhausted. All attempt exceptions
+	 * are added as suppressed exceptions to the final exception.
+	 */
+	@Override
```

Comment:
Is this Javadoc different than the Javadoc declared in the interface?

If not, you can remove the duplicated Javadoc.

Reply:
Yes, it's different. It doesn't mention "according to the retry policy configured at the implementation level" and also has more details about suppressed exceptions.